### PR TITLE
[BIOMAGE-1597] - Worker changes to support Enrichr

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-1597

#### Link to staging deployment URL 

In UI PR

#### Links to any Pull Requests related to this

UI : https://github.com/hms-dbmi-cellenics/ui/pull/602
Worker : https://github.com/hms-dbmi-cellenics/worker/pull/231

#### Anything else the reviewers should know about the changes here

Added `genesOnly` flag to return genes only.

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/hms-dbmi-cellenics/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [X] Photo of a cute animal attached to this PR

image

### CODE REVIEW

**Review by ogibson — APPROVED** [review]

File: r/R/differential_expression.R [hunk 1]
```diff
@@ -59,24 +59,36 @@ runDE <- function(req, data) {
...
-  pathwayAnalysis <- req$body$pathwayAnalysis
+  genes_only <- FALSE
+  
   order_by <- pagination$orderBy
   order_decreasing <- pagination$orderDirection == "DESC"
   offset <- pagination$offset
   limit <- pagination$limit
   filters <- pagination$filters
 
+  if ("genesOnly" %in% names(pagination)) {
```
Comment (ogibson) [inline r/R/differential_expression.R]: I dont *love* that this is inside pagination but that's a tiny detail

File: r/R/differential_expression.R [hunk 2]
```diff
@@ -59,24 +59,36 @@ runDE <- function(req, data) {
...
+    n_genes = min(limit, nrow(result))
+    
+    result <- result[order(result[, order_by], decreasing = order_decreasing), ]
+
+    new_result = list(
+      gene_names=result$gene_names[1:n_genes],
+      gene_id=result$Gene[1:n_genes]
+    )
+
+    result <- list(gene_results=new_result,full_count=n_genes)
```
Comment (ogibson) [inline r/R/differential_expression.R]:
```suggestion
    gene_results = list(
      gene_names=result$gene_names[1:n_genes],
      gene_id=result$Gene[1:n_genes]
    )

    result <- list(gene_results=gene_results,full_count=n_genes)
```

File: r/R/differential_expression.R (hunk 1 above)
Comment (aerlaut) [reply r/R/differential_expression.R]: yea it's kinda wrong. I'll put it somewhere else

File: r/R/differential_expression.R [hunk 3]
```diff
@@ -68,15 +69,27 @@ runDE <- function(req, data) {
...
+
+    result <- result[order(result[, order_by], decreasing = order_decreasing), ]
+
+    new_result <- list(
+      gene_names = result$gene_names[1:n_genes],
+      gene_id = result$Gene[1:n_genes]
+    )
+
+    result <- list(gene_results = new_result, full_count = n_genes)
+    return(result)
```
Comment (ogibson) [inline r/R/differential_expression.R]:
```suggestion
    gene_results <- list(
      gene_names = result$gene_names[1:n_genes],
      gene_id = result$Gene[1:n_genes]
    )

    result <- list(gene_results = gene_results, full_count = n_genes)
    return(result)
```

File: r/R/differential_expression.R (hunk 1 above)
Comment (aerlaut) [reply r/R/differential_expression.R]: Moved to body

**Review by ogibson — APPROVED** [review]
Sorry Agi, I thought this was already approved!

**Pull request merged**